web/src/main/java/se/inera/intyg/webcert/web/service/notification/NotificationRedeliveryStatusUpdateCreatorService.java

```diff
@@ -62,64 +55,49 @@
+        throws IOException, ModuleException, ModuleNotFoundException, TemporaryException, JAXBException {
         if (containsMessage(redelivery)) {
-            return createStatusUpdateFromExistingMessage(redelivery, event);
+            return getStatusUpdateXmlFromNotificationRedelivery(redelivery);
         }
 
-        return createStatusUpdateFromEvent(event);
+        return createStatusUpdateXmlFromEvent(event);
     }
 
     private boolean containsMessage(NotificationRedelivery redelivery) {
         return redelivery.getMessage() != null;
     }
 
-    private CertificateStatusUpdateForCareType createStatusUpdateFromExistingMessage(NotificationRedelivery redelivery, Handelse event)
-        throws IOException {
-        final NotificationRedeliveryMessage redeliveryMessage = getRedeliveryMessage(redelivery);
-
-        final var certificate = redeliveryMessage.getCert();
-        final var messagesSent = createMessages(redeliveryMessage.getSent());
-        final var messagesReceived = createMessages(redeliveryMessage.getReceived());
-        final var handledBy = NotificationRedeliveryUtil.getIIType(new HsaId(), event.getHanteratAv(), HSA_ID_OID);
-        final var statusUpdateEvent = NotificationRedeliveryUtil.getEventV3(event.getCode(), event.getTimestamp(), event.getAmne(),
-            event.getSistaDatumForSvar());
-
-        final var statusUpdate = new CertificateStatusUpdateForCareType();
-        statusUpdate.setSkickadeFragor(messagesSent);
-        statusUpdate.setMottagnaFragor(messagesReceived);
-        statusUpdate.setRef(redeliveryMessage.getReference());
-        statusUpdate.setIntyg(certificate);
-        statusUpdate.setHanteratAv(handledBy);
-        statusUpdate.setHandelse(statusUpdateEvent);
-
-        return statusUpdate;
+    private String getStatusUpdateXmlFromNotificationRedelivery(NotificationRedelivery redelivery) throws IOException {
+        return objectMapper.readValue(redelivery.getMessage(), String.class);
     }
 
-    private CertificateStatusUpdateForCareType createStatusUpdateFromEvent(Handelse event)
-        throws TemporaryException, ModuleNotFoundException, IOException, ModuleException {
+    private String createStatusUpdateXmlFromEvent(Handelse event)
+        throws TemporaryException, ModuleNotFoundException, IOException, ModuleException, JAXBException {
+
+        CertificateStatusUpdateForCareType statusUpdate;
+
         if (isDeletedEvent(event)) {
             final var careProvider = hsaOrganizationsService.getVardgivareInfo(event.getVardgivarId());
             final var careUnit = hsaOrganizationsService.getVardenhet(event.getEnhetsId());
             final var personInfo = hsaPersonService.getHsaPersonInfo(event.getCertificateIssuer()).get(0);
-            return certificateStatusUpdateForCareCreator.create(event, careProvider, careUnit, personInfo);
+            statusUpdate =  certificateStatusUpdateForCareCreator.create(event, careProvider, careUnit, personInfo);
```

Comment:
If you can, avoid mutable variables and multiple branches. It requires more to read the code because the reader have to keep track of variables that can be updated and also which branch of a statement was executed.

By just returning here instead of storing in statusUpdate you can both get rid of the statusUpdate variable and the else-clause. 

I believe you had to change this from the original code when dealing with the event time. Sorry for not noticing before.